udf/fenix_build_to_datetime/udf.sql

```diff
@@ -6,18 +6,23 @@ CREATE OR REPLACE FUNCTION udf.fenix_build_to_datetime(app_build STRING) AS (
   THEN
     -- Ideally, we would use PARSE_DATETIME, but that doesn't support
     -- day of year (%j) or the custom single-character year used here.
-    DATETIME_ADD(
-      DATETIME(
-        2018 + SAFE_CAST(
-          SUBSTR(app_build, 1, 1) AS INT64
-        ), -- year
-        1, -- placeholder month
-        1, -- placeholder year
-        SAFE_CAST(SUBSTR(app_build, 5, 2) AS INT64),
-        SAFE_CAST(SUBSTR(app_build, 7, 2) AS INT64),
-        0  -- seconds is always zero
+    IF(
+      SUBSTR(app_build, 5, 2) < "24"
```

Comment:
This is actually not a problem in the dataset I'm currently querying, but might as well validate this too.

Reply:
I'd prefer to move this IF into the CASE to avoid the extra level of nesting.

```
CASE
WHEN LENGTH(app_build) = 8 AND SUBSTR(app_build, 5, 2) < "24" AND SUBSTR(app_build, 7, 2) < "60"
THEN ...
WHEN LENGTH(app_build) = 10
THEN ...
ELSE null
END
```